src/Core/Migration/V6_7/Migration1728040169AddPrimaryOrderDelivery.php

```diff
@@ -0,0 +1,53 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Migration\V6_7;
+
+use Doctrine\DBAL\Connection;
+use Shopware\Core\Framework\Log\Package;
+use Shopware\Core\Framework\Migration\MigrationStep;
+
+/**
+ * @internal
+ */
+#[Package('checkout')]
+class Migration1728040169AddPrimaryOrderDelivery extends MigrationStep
+{
+    public function getCreationTimestamp(): int
+    {
+        return 1728040169;
+    }
+
+    public function update(Connection $connection): void
+    {
+        // No foreign key set both from order -> (primary) order delivery on
+        // purpose so the DAL can handle the circular reference. We have a similar situation with the order and order
+        // address.
+
+        if (!$this->columnExists($connection, 'order', 'primary_order_delivery_id')) {
+            $connection->executeStatement(
+                'ALTER TABLE `order`
+                ADD COLUMN `primary_order_delivery_id` BINARY(16) NULL DEFAULT NULL,
+                ADD COLUMN `primary_order_delivery_version_id` BINARY(16) NULL DEFAULT NULL,
+                ADD UNIQUE INDEX `uidx.order.primary_order_delivery` (`id`, `version_id`, `primary_order_delivery_id`);'
+            );
+
+            $connection->executeStatement(
```

Comment:
should we do this in batches? I could imagine, that this could bring down the MySQL server, if there are millions of orders in the DB